src/main/scala/subsystem/Cluster.scala

```diff
@@ -0,0 +1,230 @@
+package freechips.rocketchip.subsystem
+
+import chisel3._
+import chisel3.util._
+
+import org.chipsalliance.cde.config.{Field, Parameters}
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.interrupts._
+import freechips.rocketchip.prci._
+import freechips.rocketchip.tile.{RocketTile, NMI, TraceBundle}
+import freechips.rocketchip.subsystem._
+import freechips.rocketchip.tilelink._
+import freechips.rocketchip.devices.debug.{TLDebugModule}
+import freechips.rocketchip.devices.tilelink._
+import freechips.rocketchip.util._
+import scala.collection.immutable.ListMap
+
+case class ClustersLocated(loc: HierarchicalLocation) extends Field[Seq[CanAttachCluster]](Nil)
+
+case class ClusterParams(
+  val clusterId: Int,
+  val clockSinkParams: ClockSinkParameters = ClockSinkParameters()
+) extends ElementParams {
+  val baseName = "cluster"
+  val uniqueName = s"${baseName}_$clusterId"
+  def instantiate(crossing: ElementCrossingParamsLike, lookup: LookupByClusterIdImpl)(implicit p: Parameters): Cluster = {
+    new Cluster(this, crossing.crossingType, lookup)
+  }
+}
+
+class Cluster(
+  val thisClusterParams: ClusterParams,
+  crossing: ClockCrossingType,
+  lookup: LookupByClusterIdImpl)(implicit p: Parameters) extends BaseElement(crossing)(p)
+    with Attachable
+    with HasConfigurableTLNetworkTopology
+    with InstantiatesElements
+    with HasElements
+{
+  val busContextName = thisClusterParams.baseName
+  lazy val clusterId = thisClusterParams.clusterId
+  lazy val location = InCluster(clusterId)
+
+  lazy val allClockGroupsNode = ClockGroupIdentityNode()
+
+  val csbus = tlBusWrapperLocationMap(SBUS) // like the sbus in the base subsystem
+  val ccbus = tlBusWrapperLocationMap(CBUS) // like the cbus in the base subsystem
+  val cmbus = tlBusWrapperLocationMap.lift(MBUS).getOrElse(csbus)
```

Comment:
For each cluster, they should have these buses?
or we can let user directly query bus via location API.

Reply:
I will try to remove what I can here.